src/transformers/models/recurrent_gemma/modeling_recurrent_gemma.py

```diff
@@ -895,37 +896,31 @@ def forward(
             hidden_states=outputs.hidden_states,
         )
 
-    # Ignore copy
+    # Ignore copy (no past_key_values, no inputs_embeds)
     def prepare_inputs_for_generation(
-        self, input_ids, attention_mask=None, inputs_embeds=None, cache_position=None, use_cache=None, **kwargs
+        self, input_ids, attention_mask=None, cache_position=None, use_cache=None, position_ids=None, **kwargs
     ):
-        position_ids = kwargs.get("position_ids", None)
+        if cache_position.shape[0] == 1 and cache_position[0] == 0:
+            raise ValueError(
+                "Recurrent gemma doesn't work with an empty text prompt (i.e. only BOS token), as a minimum of two "
+                "tokens are needed to detect prefill. Please provide a non-empty text prompt."
+            )
```

Comment:
We have to pick one, because we need to detect we're doing prefill in the fwd pass:
1. the model doesn't work with empty text prompt (we detect prefill with `cache_position.shape[0] != 1`)
2. the model doesn't compile (we detect prefill with `cache_position[0] == 0`)
3. we add a new flag in the forward pass (in the model class, before transforming `attention_mask`, we can detect prefill)

None of them seems to me like a clear win, so I went with the option that was already in the code :) This exception exposes the limitation to the user.


1. 
2.
3.

Reply:
interesting. Won't this be an issue for generate compile as this function would be compiled -> input dependent branches?